blob type, iOS, bugfixes

Lasibian opened the pull request:
Added blob data type.

The nextgen compiler (XE7 with target iOS) does not support the Contnrs unit. Therefore I've changed TObjectList to TObjectList<TOSCPacket>. Also some adjustments were needed for the zero based strings.

The current version of Indy requires a cast from or to TBytes and TIdBytes.

Bugfix: when adding an integer or float argument and immediately reading it back, the byte order is reversed. Example:

``` delphi
var
  msg: TOSCMessage;
begin
  msg := TOSCMessage.Create('/test/x');
  msg.AddInteger(123);
  if msg.ArgumentAsInt[0] <> 123 then
    Writeln('fail ', msg.ArgumentAsInt[0]);
end;
```

In order to workaround that problem, I've changed the code, so that the byte order adjustment happens when reading or writing the arguments. This change should not affect existing code.

Commit:
blob type, iOS, bugfixes

Added blob type, changes for iOS
Bugfix: when adding an integer or float argument and immediately reading
it back, the byte order is reversed

joreg commented:
helo @Lasibian,

sorry for the late reply and thanks for your contribution. only: would you mind providing your changes again but as separate pull requests? i'd prefer merging smaller more readable commits. to my understanding at the moment your pullrequest mixes 4 changes:
- add blob type
- fix for iOS/nextgen
- fix for zero based strings
- byteorder

hope that would work for you.

Commit:
GetArgumentAsString: only trim #0

Changed trim so it only removes trailing #0 at the end, not all white
space characters

Commit:
iOS bugfix

GetTypeTag: Bugfix for zero based string handling

Commit:
Bugfixes

Commit:
Implemented Double Argument
